### PR TITLE
Added percentage and year to the result, refactored sorting function

### PR DESCRIPTION
**Pull request by kamyarsedghi**
- Added percentage to the results
- Added year beside each month in the results
- Refactored the sorting function for the values with "month,year" template
- Fixed showing days of 0 for the rental sessions that are rented in the same day

### CODE REVIEW

File: src/car/car.service.ts [hunk 1]
```diff
@@ -93,21 +98,50 @@ export class CarService {
...
+                    return 1;
+                } else {
+                    if (monthName.indexOf(monthA) < monthName.indexOf(monthB)) {
+                        return -1;
+                    } else if (monthName.indexOf(monthA) > monthName.indexOf(monthB)) {
+                        return 1;
+                    } else {
+                        return 0;
+                    }
+                }
```
Comment (a-lakiza) [inline src/car/car.service.ts]: again every `else` is useless

File: src/car/car.service.ts [hunk 2]
```diff
@@ -2,6 +2,8 @@ import { BadRequestException, Injectable } from '@nestjs/common';
 import { ReservationService } from 'src/reservation/reservation.service';
 import { DatabaseService } from 'src/utils/database/database.service';
 import { DateQueryDto } from './dto/dateQuery.dto';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const ObjectsToCsv = require('objects-to-csv');
```
Comment (a-lakiza) [inline src/car/car.service.ts:8]: why don't you use `import` there?

File: src/car/car.service.ts [hunk 3]
```diff
@@ -93,21 +98,50 @@ export class CarService {
...
+                    return {
+                        car_id: key,
+                        ...value,
+                    };
+                }),
+            };
+        });
+
+        const csv = new ObjectsToCsv(toFile);
+        await csv.toDisk('./carsUsage.csv');
```
Comment (a-lakiza) [inline src/car/car.service.ts]: I want to have both options csv and json

File: src/car/car.service.ts (hunk 1 above)
Comment (kamyarsedghi) [reply src/car/car.service.ts]: I apologize for using useless elses, however I have refactored it by using ternary operators.

File: src/car/car.service.ts (hunk 2 above)
Comment (kamyarsedghi) [reply src/car/car.service.ts:8]: Using import for this module would result in the error show below:
`TypeError: objects_to_csv_1.default is not a constructor`
After some googling, people suggested that using const would work.
Do you have any solutions for this?

File: src/car/car.service.ts (hunk 3 above)
Comment (kamyarsedghi) [reply src/car/car.service.ts]: Feature added.

File: src/car/car.service.ts (hunk 2 above)
Comment (a-lakiza) [reply src/car/car.service.ts:8]:
```suggestion
import * as ObjectsToCsv from 'objects-to-csv';
```

Comment (a-lakiza) [reply src/car/car.service.ts:8]: did you try this one?

Comment (kamyarsedghi) [reply src/car/car.service.ts:8]: Yes absolutely.
This was the error:
image

Comment (kamyarsedghi) [reply src/car/car.service.ts:8]: This is from the library itself.

image
image

Comment (a-lakiza) [reply src/car/car.service.ts:8]: okay

**Review by a-lakiza — APPROVED** [review]

**Pull request merged**